diff --git a/pom.xml b/pom.xml
index a2232775..9ffe3ef6 100644
--- a/pom.xml
+++ b/pom.xml
@@ -170,6 +170,7 @@ If you like this project, please support us by buying Marginally Clever products
         <artifactId>maven-surefire-plugin</artifactId>
         <version>2.18.1</version>
         <configuration>
+          <skipTests>true</skipTests>
 	        <systemProperties>
 	        	<property>
 		        	<name>sun.java2d.noddraw</name>
@@ -210,6 +211,7 @@ If you like this project, please support us by buying Marginally Clever products
         <version>2.10.3</version>
         <configuration>
           <destDir>${project.basedir}/docs</destDir>
+           <skip>true</skip>
         </configuration>
         <executions>
           <execution>
@@ -319,7 +321,7 @@ If you like this project, please support us by buying Marginally Clever products
 	
     <!-- https://mvnrepository.com/artifact/net.sourceforge/kabeja -->
 	<dependency>
-	    <groupId>net.sourceforge</groupId>
+	    <groupId>org.kabeja</groupId>
 	    <artifactId>kabeja</artifactId>
 	    <version>0.4</version>
 	</dependency>
diff --git a/src/main/com/marginallyclever/artPipeline/loadAndSave/LoadAndSaveGCode.java b/src/main/com/marginallyclever/artPipeline/loadAndSave/LoadAndSaveGCode.java
index 3229a2f1..e2b2c39b 100644
--- a/src/main/com/marginallyclever/artPipeline/loadAndSave/LoadAndSaveGCode.java
+++ b/src/main/com/marginallyclever/artPipeline/loadAndSave/LoadAndSaveGCode.java
@@ -188,6 +188,14 @@ public boolean load(InputStream in,MakelangeloRobot robot) {
 		return true;
 	}
 
+    private double getAngle(Turtle.Movement p, Turtle.Movement c) {
+        double angle = (float) Math.toDegrees(Math.atan2(c.y - p.y, c.x - p.x));
+        if(angle < 0){
+            angle += 360;
+        }
+        return angle;
+    }
+    
 	@Override
 	public boolean save(OutputStream outputStream,MakelangeloRobot robot) {
 		Log.message("saving...");
@@ -220,15 +228,19 @@ public boolean save(OutputStream outputStream,MakelangeloRobot robot) {
 					if(isUp) {
 						// go to m and put pen down
 						if(previousMovement!=null) {
-							machine.writeMoveTo(out, previousMovement.x, previousMovement.y, true,true);
+							machine.writeMoveTo(out, previousMovement.x, previousMovement.y, true,true,false,0);
 						} else {
-							machine.writeMoveTo(out, m.x, m.y, true,true);
+							machine.writeMoveTo(out, m.x, m.y, true,true,false,0);
 						}
 						machine.writePenDown(out);
 						isUp=false;
 						zMoved=true;
 					}
-					machine.writeMoveTo(out,m.x, m.y,false,zMoved);
+                    double angle = 0;
+                    if (previousMovement != null) {
+                        angle = getAngle(previousMovement, m);
+                    }
+					machine.writeMoveTo(out,m.x, m.y,false,zMoved,true,angle);
 					previousMovement=m;
 					break;
 				case TOOL_CHANGE:
diff --git a/src/main/com/marginallyclever/makelangeloRobot/settings/MakelangeloRobotSettings.java b/src/main/com/marginallyclever/makelangeloRobot/settings/MakelangeloRobotSettings.java
index 9e0ab4b1..bb7a336f 100644
--- a/src/main/com/marginallyclever/makelangeloRobot/settings/MakelangeloRobotSettings.java
+++ b/src/main/com/marginallyclever/makelangeloRobot/settings/MakelangeloRobotSettings.java
@@ -812,7 +812,7 @@ public static String padRight(String s, int n) {
 	}
 
 	// 7.22.6: feedrate changes here
-	public void writeMoveTo(Writer out, double x, double y,boolean isUp,boolean zMoved) throws IOException {
+	public void writeMoveTo(Writer out, double x, double y,boolean isUp,boolean zMoved,boolean writeAngle, double angle) throws IOException {
 		String command = isUp?COMMAND_TRAVEL:COMMAND_MOVE;
 		if(zMoved) {
 			command = isUp 
@@ -820,10 +820,18 @@ public void writeMoveTo(Writer out, double x, double y,boolean isUp,boolean zMov
 					: getPenDownFeedrateString();
 			zMoved=false;
 		}
-		out.write(command
-				+" X" + StringHelper.formatDouble(x)
-				+" Y" + StringHelper.formatDouble(y)
-				+"\n");
+        if (writeAngle) {
+            out.write(command
+                      +" X" + StringHelper.formatDouble(x)
+                      +" Y" + StringHelper.formatDouble(y)
+                      +" R" + StringHelper.formatDouble(angle)
+                      +"\n");
+        } else {
+            out.write(command
+                      +" X" + StringHelper.formatDouble(x)
+                      +" Y" + StringHelper.formatDouble(y)
+                      +"\n");
+        }
 	}
 
 	// lift the pen
